modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/ResponsiveViewer.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+
+import * as R from 'ramda';
+import React from 'react';
+import type {Node} from 'react';
+import {scaleLinear} from 'd3-scale';
+import {withParentSize} from '@visx/responsive';
+import type {Vector2} from '../../vector';
+
+type Props = {
+  parentWidth: number,
+  parentHeight: number,
+  mouseDown: Vector2 => void,
+  mouseMove: Vector2 => void,
+  mouseUp: Vector2 => void,
+  mouseLeave: Vector2 => void,
+  children: Node
+};
+
+const ResponsiveViewer = ({
+  parentWidth,
+  parentHeight,
+  mouseDown,
+  mouseMove,
+  mouseUp,
+  mouseLeave,
+  children,
+}: Props) => {
```

Comment:
Is this a way of declaring prop types/default props? If so, I like it!

Reply:
Yes, but it's a syntax specific to flow. Typescript is more popular nowadays.